extensions/lifecycle/CircuitBreakerGroup.js

```diff
@@ -0,0 +1,317 @@
+                [...this.circuitBreakers.transition.location[location], circuitBreaker] : [circuitBreaker];
+        }
+    }
+
+    addExpirationCircuitBreaker(circuitBreaker, topic, location) {
+        if (!topic && !location) {
+            this.circuitBreakers.expiration.global.push(circuitBreaker);
+        }
+        if (topic) {
+            this.circuitBreakers.expiration.topic[topic] =  this.circuitBreakers.expiration.topic[topic] ?
+                [...this.circuitBreakers.expiration.topic[topic], circuitBreaker] : [circuitBreaker];
+        }
+        if (location) {
+            this.circuitBreakers.expiration.location[location] =  this.circuitBreakers.expiration.location[location] ?
+                [...this.circuitBreakers.expiration.location[location], circuitBreaker] : [circuitBreaker];
+        }
+    }
+
+    exportCircuitBreakerMetric(circuitBreaker, metricsSuffix) {
+        let metricsName = 'lifecycle_bucket_processor';
+        if (metricsSuffix) {
+            metricsName += `_${metricsSuffix}`;
+        }
+        metricsName += `_${this._latestBreakerId}`;
+        this._latestBreakerId += 1;
+        startCircuitBreakerMetricsExport(
+            circuitBreaker,
+            metricsName
+        );
+    }
+
+    addCircuitBreaker(probeConf, query, isTransition, isExpiration, topic, location, metricsSuffix) {
+        const circuitBreaker = new CircuitBreaker({
+            ...this._cbGlobalConf,
+            probes: [{
+                ...probeConf,
+                query,
+            }],
+        });
+        this.exportCircuitBreakerMetric(circuitBreaker, metricsSuffix);
+        if (isTransition) {
+            this.addTransitionCircuitBreaker(circuitBreaker, topic, location);
+        }
+        if (isExpiration) {
+            this.addExpirationCircuitBreaker(circuitBreaker, topic, location);
+        }
+    }
+
+    /**
+     * Checks if circuit breakers appropriate to our action are triggered
+     * @param {string} worfklowType transition or expiration
+     * @param {string} location location name of object
+     * @param {string} topic topic we want to push to
+     * @returns {bool} true if it should circuit break
+     */
+    tripped(worfklowType, location, topic) {
+        const breakers = [];
+        if (!worfklowType || worfklowType === 'expiration') {
+            if (location && this.circuitBreakers.expiration.location[location]) {
+                breakers.push(
+                    ...this.circuitBreakers.expiration.location[location],
+                );
+            }
+            if (topic && this.circuitBreakers.expiration.topic[topic]) {
+                breakers.push(
+                    ...this.circuitBreakers.expiration.topic[topic],
+                );
+            }
+            breakers.push(
+                ...this.circuitBreakers.expiration.global,
+            );
```

Comment:
nit: this could be deduplicated (same thing on `.transition` & `.expiration`, either with a loop or a function